### ISSUE TITLE
Wrong working directory for recipes inside imported justfile

### ISSUE THREAD

**Issue by Ziqi-Yang** [casey/just#1978]
Directory Structure
```bash
/tmp/experiment took 5s 
❯ tree .
.
└── just
    ├── a.just
    └── b.just
```
File content:
```bash
❯ cat ./just/a.just
#!/usr/bin/env -S just --working-directory . --justfile

import 'b.just'

a: b
  pwd
❯ cat ./just/b.just
b:
  pwd
```
When I run `./just/a.just`, it produces:
```
/tmp/experiment 
❯ ./just/a.just
pwd
/tmp/experiment/just
pwd
/tmp/experiment
```

**Comment by Ziqi-Yang** [comment]
This is the source: https://github.com/casey/just/blob/610aa0c52cf8c3d20a79ee641bb9f799ca3027fc/src/recipe.rs#L130  
It's intentionally set, but I think it's arguable to make the working directory of the imported recipe different than the command line argument. It may be better to provide a option for `import` clause or something else.

**Comment by Ziqi-Yang** [comment]
I see there is `[no-cd]` clause, but it's for single recipe, but not for the whole justfile. And it we have complicated function call hierarchy like `python name: (jpg-copy-template name "python") && (jpg-replace-builtin name)`  then we need to put `[no-cd]` attribute to all of the recipes here.

**Comment by Ziqi-Yang** [comment]
> This is the source:
> 
> https://github.com/casey/just/blob/610aa0c52cf8c3d20a79ee641bb9f799ca3027fc/src/recipe.rs#L130
> 
> It's intentionally set, but I think it's arguable to make the working directory of the imported recipe different than the command line argument. It may be better to provide a option for `import` clause or something else.

Also, I don't think using `self.working_directory` makes much sense, since it's neither the path that the user is at (maybe is, when the user is at the directory that holds the entry justfile), nor the path that the imported justfile is at.

**Comment by casey** [comment]
Thanks for opening this issue! I am honestly pretty confused why I wrote the code this way, I think it might just have been an oversight.

Working directories for modules and imports has been surprisingly tricky. I'm going to poke around and see if this can just be changed.

**Comment by casey** [comment]
Modules and imports have different behavior. Recipes in imports run in the same directory as the parent, but they should respect `search.working_directory`.

**Comment by casey** [comment]
Fixed in #2056. Recipes now have a `file_depth` and a `submodule_depth`, `import`s increase `file_depth` but not `submodule_depth`, whereas `mod`s increase both. The test in `Recipe::working_directory` now use `submodule_depth`, so imports in the root of a justfile will take `--working-directory` into account.
